[ADD] PLM: approvals doc (#5348)
odoo/documentation: content/applications/inventory_and_mrp/manufacturing/plm/approvals.rst

   the `New Product Introduction` ECO type. Approvals from the quality and marketing teams are not
   required to apply changes to the |ECO| because their :guilabel:`Approval Type` is set to
   :guilabel:`Approves, but the approval is optional` and :guilabel:`Comments only`, respectively.

   .. image:: approvals/approvers.png
      :align: center
      :alt: Set an approver that "Is required to approve" ECOs in the "Validated" stage.

Manage approvals
================

Approvers can easily track their to-do approvals by navigating to the :menuselection:`PLM app` and
looking at the ECO type card, which shows the count of open tasks assigned to them. Here's what each
button on the ECO type card does:

#. :guilabel:`X Engineering Changes` button displays a count of in-progress |ECOs| of this ECO type.
   Clicking the button opens the Gantt view of the :guilabel:`Engineering Change Orders` page.

#. :guilabel:`My Validations` displays a count of |ECOs| that the approver must accept or reject.
   Clicking on this button displays |ECOs| pending approval or rejected (marked with the red
   :guilabel:`Blocked` blocked state).

> #. :guilabel:`All Validations` displays a count of |ECOs| a verification stage pending approval from
>    any approvers. Clicking on the button displays |ECOs| pending approval from or rejected by any
>    approver.

Author: ```suggestion
The  :guilabel:`All Validations` button shows the count of |ECOs| awaiting approval or rejected by any approver. Clicking it reveals these pending |ECOs|.```
```

#. :guilabel:`To Apply` displays a count of |ECOs| the user needs to apply changes to. Clicking on
   the button displays all the |ECOs| to approve and apply changes to in the verification stage.
   |ECOs| marked with the green :guilabel:`Done` stage have already been approved, and the user just
   needs to click on the |ECO| to enter the form view and click the :guilabel:`Apply Changes`
   button.

.. image:: approvals/validation-overview.png
   :align: center
   :alt: Display count of validations to-do and buttons to open filtered list of ECOs.

Approve ECOs
------------

Navigate to an |ECO| in a verification stage while logged in as the assigned approval to see the
:guilabel:`Approve`, :guilabel:`Reject`, and :guilabel:`Apply Changes` button. To approve the |ECO|
and apply the changes onto the production :abbr:`BoM (Bill of Materials)`, click :guilabel:`Approve`
then :guilabel:`Apply Changes`. Note that the :guilabel:`Apply Changes` button will not work unless
the :guilabel:`Approve` button was pressed first. Additionally, the chatter will log the history of

> the buttons pressed.

Reviewer: apply changets button will not work until all approvers have approvede. (When there are multiple approvers)

> the :guilabel:`Approve` button was pressed first. Additionally, the chatter will log the history of
> the buttons pressed.

Reviewer 2: ```suggestion
the :guilabel:`Approve` button was clicked first. Additionally, the chatter will log the history of
the buttons clicked.
```

.. warning::
   When the :guilabel:`Approval Type` is **not** set to :guilabel:`Is required to approve`, approval

>    from the associated user is not needed before applying changes with the the :guilabel:`Apply
>    Changes` button. Thus, the :guilabel:`Apply Changes` **will work** without the requiring the
>    :guilabel:`Approve` button to be pressed first. pressed before it.

Reviewer 2: ```suggestion
   from the associated user is not needed before applying changes with the :guilabel:`Apply
   Changes` button. Thus, the :guilabel:`Apply Changes` button **will work** without requiring the
   :guilabel:`Approve` button to be clicked first.
```

Automated activities
--------------------

When an |ECO| is moved to a verification stage, a planned activity is automatically created for
assigned approvers to review the |ECO|. Approvers receive a notification in their activities inbox,

> accessible through the :guilabel:`🕘 (clock)` icon at the top of the page. In to-do task list, the
> :guilabel:`Engineering Change Order (ECO)` notification displays the amount of :guilabel:`Late`,
> :guilabel:`Today`'s, and :guilabel:`Future` activities. Clicking on each of these buttons shows a

Reviewer 2: ```suggestion
accessible through the :guilabel:`🕘 (clock)` icon at the top of the page. In the to-do task list, the
:guilabel:`Engineering Change Order (ECO)` notification displays the number of 
activities marked :guilabel:`Late`,
:guilabel:`Today`, and :guilabel:`Future`. Clicking on each of these buttons shows a
```

filtered Gantt view of the respective |ECOs|.

.. example::

>    Scheduled activities are shown in the :guilabel:`🕘 (clock)` icon, with `5` |ECOs| pending

Reviewer 2: ```suggestion
   Scheduled activities are shown as a number on the :guilabel:`🕘 (clock)` icon, with `5` |ECOs| pending
```

   approval :guilabel:`Today`. Currently, there are `0` :guilabel:`Late` or :guilabel:`Future`
   |ECOs|.

    .. image:: approvals/todo-list.png
       :align: center
       :alt: Show scheduled approvals notifications for the user.

> Click each pending |ECO|, and a *planned activity* for :guilabel:`ECO Approval` is recorded in the

Reviewer 2: ```suggestion
By clicking a pending |ECO|, a *planned activity* for :guilabel:`ECO Approval` is recorded in the
```

> chatter. Clicking on the :guilabel:`i (Info)` icon to view additional information, including the

Reviewer 2: ```suggestion
chatter. Click on the :guilabel:`i (Info)` icon to view additional information, including the
```

approval's :guilabel:`Created` date, the approver :guilabel:`Assigned to` it, and the due date.

.. image:: approvals/planned-activity.png
   :align: center
   :alt: Show additional details of the planned ECO approval.

Follow-up activities
~~~~~~~~~~~~~~~~~~~~

> When |ECOs| are rejected, tasks need be assigned project members for required modifications before

Reviewer 2: ```suggestion
When |ECOs| are rejected, tasks need to be assigned to project members for required modifications before
```
